.circleci/scan_artifacts.ts

```diff
@@ -1,20 +1,32 @@
 import * as execa from 'execa';
 import * as path from 'path';
-const DEFAULT_SEARCH_FOLDER = path.normalize(path.join(__dirname, '..'));
+import { ARTIFACT_STORAGE_PATH_ALLOW_LIST } from '../scripts/artifact-storage-path-allow-list';
+const REPO_FOLDER = path.normalize(path.join(__dirname, '..', '..'));
 export const hasMatchingContentInFolder = (
   patterns: string[],
-  folder = DEFAULT_SEARCH_FOLDER,
-  excludeFolder = '{*/node_modules,.git}',
+  folder: string,
+  excludeFolder = '{node_modules,.cache,.git,\.cache,\.git}',
 ): boolean => {
+  console.log("Scanning folder:", folder);
   const patternParam = patterns.reduce<string[]>((acc, v) => [...acc, '-e', v], []);
+
+  let actualFolder = folder;
+  if(folder.startsWith("~/") || folder.startsWith("~\\")){
+    actualFolder = folder.replace("~", REPO_FOLDER);
+  }
```

Comment:
ah, now I understand.
I'd replace `~` with something more self explanatory like `[repo_folder]` or some other more verbose token/placeholder.
`~` is reserved for user's home in linux shells.

Reply:
in CircleCI it is also the working directory (both linux & windows)

Reply:
we cannot include special characters like [repo_folder] in the config.yaml paths

Reply:
paths in the config.yaml are 1 to 1 with the allow-list ; and circleci can't interpret [repo_folder], so we'd also have to implement an override script to replace [repo_folder] in config.yaml with ~ or similar

Reply:
Okay, I think I understand the constraints now. Circle CI must be then doing similar trick internally on windows (~ is meaningless in windows shell I recall).
Can you change naming or add comments ? or even extract this logic to `normalizeFolderPath` function and mention those things in the comments?
If REPO_FOLDER is absolute path we can name it accordingly ?

Reply:
sure; that make sense, ill update it and add comments